ably/realtime_channel.go

```diff
@@ -585,25 +585,42 @@ func (em ChannelEventEmitter) OffAll() {
 }
 
 // Publish publishes a single message to the channel with the given event name and message payload.
-// A callback may optionally be passed in to this call to be notified of success or failure of the operation.
-// When publish is called with this client library, it won't attempt to implicitly attach to the channel,
-// so long as transient publishing is available in the library. Otherwise, the client will implicitly attach (RTL6i).
-//
-// This implicitly attaches the channel if it's not already attached. If the
-// context is canceled before the attach operation finishes, the call
-// returns with an error, but the operation carries on in the background and
-// the channel may eventually be attached and the message published anyway.
 func (c *RealtimeChannel) Publish(ctx context.Context, name string, data interface{}) error {
 	return c.PublishMultiple(ctx, []*Message{{Name: name, Data: data}})
 }
 
+// PublishAsync is the same as Publish except instead of blocking it calls onAck after either:
+// * The message is successfully acknowledged so onAck is called with a nil error
+// * The publish fails so onAck is called with the appropriate error
```

Comment:
- PublishAsync is a non-blocking Publish that accepts `onAck` callback fn as an extra arg,
- onAck is called with error as nil/non-nil specifying success/failure respectively

Reply:
Can you update doc for the same for `publishMultipleAsync`?

Reply:
How about
```
// PublishAsync is the same as Publish except instead of blocking it calls onAck
// with nil if the publish was successful or an appropriate error.
```

Reply:
// PublishAsync is the same as Publish except instead of blocking it calls onAck
// with nil if the publish was successful or with error value if publish fails.
//
// Important Note - code inside onAck will block internal client, it should be written in a way that it shouldn't be blocking and shouldn't take too much time to execute.